design-documents/graph-ql/coverage/search.md

```diff
@@ -0,0 +1,79 @@
+# Queries
+
+```graphql
+type Query {
+    multiSearch(phrase: String!, productSize: Int = 10): MultiSearchResponse!
+    #Filter supports multiple clauses which will be wrapped in logical AND operator
+    productSearch(phrase: String!, filter: [SearchClauseInput], sort: [ProductSearchSortInput]): ProductSearchResponse!
+}
+
+input ProductSearchSortInput
+{
+    attribute: String!
+    direction: SortEnum!
+}
+
+# If from or to fields are omitted, $gte or $lte filter will be applied
+input SearchRangeInput {
+    from: Int
```

Comment:
`from` and `to` aren't the most descriptive names here, and their value is a scalar which doesn't help a developer infer what value to pass.

Is there a name that would make this more clear? I'm guessing `from` and `to` are index offsets based on the ordered collection of results. In that case, maybe `startIndex` or `startOffset` and an equivalent for `to`.

Also not clear as a developer consuming this schema whether these offsets are inclusive or not

Reply:
Is range == price range? https://devdocs.magento.com/guides/v2.3/rest/search-endpoint.html#advanced-searches

Reply:
Price range + potentially range of any numerical values

Reply:
Seems like a price range filter would need `from` and `to` to be our `Money` type, otherwise I can't filter between $1.00 and $1.99

Reply:
> I still want to see reusable numerical input/filter.

What's the motivation for needing a reusable filter?

My concern is that the docs are always going to need to explain the various uses of `from` and `to` since they're so generic. Ideally the names of items in the schema are descriptive enough that you don't need to consult docs for common operations like a search.

IMO, we'd be taking a trade-off: less schema to maintain, but more docs and reading for API consumers.

Reply:
Starting to agree with you more here @kokoc. I took a look again at the usage of `ProductSearchSortInput`, and I guess it will always be specified right next to `attribute_code`, so generic `to`/`from` `float` will probably cover that 👍